### ISSUE TITLE
Old-frontend types

### ISSUE THREAD

**Issue by sadernalwis** [B3Pay/b3-wallet#33]
Has the ```old-frontend``` plugged out already. or still utilized?

I'm getting :
```Property 'add_wallet_canister' does not exist on type 'ActorSubclass<_SERVICE>'.```
compile errors from:
[      .add_wallet_canister(canisterPrincipal)](https://github.com/sadernalwis/b3-wallet/blob/f8a0e63118d2cc557ccec5565d6178b40bf08a98/old-frontend/src/components/Header.tsx#L150)

and;

```bash
Executing 'yarn build'
$ npx next build
   Linting and checking validity of types  ...Failed to compile.

./old-frontend/src/components/Header.tsx:149:5
Type error: 'systemActor' is possibly 'undefined'.

  147 |     }
  148 |
> 149 |     systemActor
      |     ^
  150 |       .add_wallet_canister(canisterPrincipal)
  151 |       .then(() => {
  152 |         setLoading(false)
error Command failed with exit code 1.
```

**Comment by b3hr4d** [comment]
I'm working on Beta release, and everything is changed on backend, since this was for Hackathon and everything is messed up and I don't have time to update it to use new backend. you can get the old b3_system contract from [old_main](https://github.com/B3Pay/b3-wallet/tree/old_main) and test things with that.

**Comment by sadernalwis** [comment]
> I'm working on Beta release, and everything is changed on backend, since this was for Hackathon and everything is messed up and I don't have time to update it with new backend. you can get the old b3_system contract from [old_main](https://github.com/B3Pay/b3-wallet/tree/old_main) and test things with that.

I wouldn't mind spending some time to get this done this week. 
would help if you can guide me if I hit a snag.
I'd assume a good starting point would be to start
removing references of ```old-frontend``` in the ```src``` without breaking code.?

**Comment by b3hr4d** [comment]
Thanks a lot for offering your help! It's really appreciated.

Regarding the old-frontend, Your idea of starting by removing old-frontend references in the src is a good approach.

If you run into any snags or need guidance, feel free to reach out. I'm here to help.

Also, I'm curious about your interests and experience in this area. What aspects do you enjoy working on the most, and what skills do you bring to the table? This info will help us collaborate more effectively.

Thanks again for your involvement!

**Comment by sadernalwis** [comment]
My background is VFX, Game Engines(Vulkan), Blender Plugins , C++/Python
I know its quite the different field.😊

Experience when it comes to ICP is 0. I just started on ICP 3months ago. only studied rust for this purpose.
Quite noviced in terms of wallets and dapp development. Bu ti think I'm a quick study.

We just started a Digital Assets startup based on ICP tech. And I think B3 could be used for its wallet functionality.

I'd like to learn more about CK/ICRC/NFT token standards and overall ICP architecture in depth.

Absolutely it's my pleasure to contribute to & maintain this project in parallel.

Going ahead you can tell me which areas in B3 you'd like me to focus.

I'm also quite intrigued by the Global-Wallet idea of yours.

**Comment by b3hr4d** [comment]
Given your interest in learning about ICP and your familiarity with Rust, I think you'll find the `b3_utils` library quite useful. It's designed to facilitate development on the Internet Computer, and you can find it here: https://github.com/B3Pay/b3_utils.

Working on the documentation for this library could be an excellent way for you to learn more about it while contributing. Your suggestions and improvements to the documentation would be greatly appreciated, as they can significantly enhance the library's usability for everyone.

Additionally, your involvement with the b3_utils library could also provide insights that may benefit your other projects. It's a practical way to get hands-on experience with ICP technology and understand its architecture more deeply.

Feel free to explore the library and let me know if you have any questions or need guidance. Your contributions to improving the documentation or any other aspect of the library would be highly valuable and very much welcomed.

Currently, my primary focus is on creating a store where anyone can upload their WebAssembly (WASM) applications for installation. I'm also working on a dynamic method to generate UI for these WASM applications using Candid. It's an exciting development that I believe will greatly benefit the ICP ecosystem.

Once you're more familiar with the b3_utils library and have a solid understanding of its functionalities, it would be great to collaborate on building the global wallet. Your mastery of the b3_utils library will be incredibly valuable in this endeavor.

This global wallet project is an ambitious one, and your contribution, especially with your background and newfound skills, could be instrumental in making it a reality. I'm looking forward to potentially working together on this and seeing where your journey with ICP takes you. Keep me updated on your progress with the b3_utils library, and don't hesitate to reach out if you have any questions or need assistance along the way!

**Comment by sadernalwis** [comment]
Sounds perfect.! Going to be an interesting 2024...

TODO: New Front-End, B3 Utils documentation.

WASM module uploads is  definitely going to be a game changer.

Super glad to be part of this project

I'll definitely keep you updated.

**Comment by b3hr4d** [comment]
I'm also super excited about the potential of what we're building. Thank you for being part of this project – your contribution is truly valued. Looking forward to seeing what we can achieve together in 2024! ❤️

**Comment by sadernalwis** [comment]
there's a lot of type breakage coming from ```typescript```  in ```scripts```.
would you mind if I convert them to ```javascript```?

**Comment by b3hr4d** [comment]
What is the exact problem, Is that hard to solve?

**Comment by sadernalwis** [comment]
I fixed a few. but many keeps popping, blocking me from getting a clean build.

![Screenshot from 2023-12-20 21-18-55](https://github.com/B3Pay/b3-wallet/assets/44372160/84cc353c-8fe7-4676-8a8c-46a85a76547c)


I could try suppressing errors and ignoring strong type checks. but that would make using ```ts``` pointless

**Comment by b3hr4d** [comment]
please solve what you can, and leave others for me to solve later.
also we might need to remove some of the files later.
Did you take a look at b3_utils library?

**Comment by sadernalwis** [comment]
got it.
yup. going through it in parallel.
waiting to get a good build to play around with it,

**Comment by sadernalwis** [comment]
getting an ```Actor not initialized``` during ```load-wasm:system```.
could it be an II issue?
I have been having an II issue [here](https://forum.dfinity.org/t/local-internet-identity-failure/18577/15?u=sadernalwis) as well.

**Comment by b3hr4d** [comment]
I guess you have other issue with root key, did you get this error `Unable to fetch root key`?

**Comment by b3hr4d** [comment]
please check this 
```js
export const initIdentity = (mainnet: boolean) => {
  const userHomeDir = os.homedir()
  const file = path.join(
    userHomeDir,
    `.config/dfx/identity/${mainnet ? "default" : "default"}/identity.pem`
  )
  const buffer = readFileSync(file)
  const key = buffer.toString("utf-8")

  return decode(key)
}
```
 do have this file or not.

**Comment by sadernalwis** [comment]
> I guess you have other issue with root key, did you get this error `Unable to fetch root key`?

will check

**Comment by sadernalwis** [comment]
> I guess you have other issue with root key, did you get this error `Unable to fetch root key`?

will check

**Comment by sadernalwis** [comment]
> please check this
> 
> ```js
> export const initIdentity = (mainnet: boolean) => {
>   const userHomeDir = os.homedir()
>   const file = path.join(
>     userHomeDir,
>     `.config/dfx/identity/${mainnet ? "default" : "default"}/identity.pem`
>   )
>   const buffer = readFileSync(file)
>   const key = buffer.toString("utf-8")
> 
>   return decode(key)
> }
> ```
> 
> do have this file or not.

yes. its there.

got the II built and deployed separately. trying to connect to that.

**Comment by b3hr4d** [comment]
I mean this file `.config/dfx/identity/${mainnet ? "default" : "default"}/identity.pem`, is it in your OS directory?

**Comment by b3hr4d** [comment]
It's different to II, this is something like private key stored in your storage!

**Comment by sadernalwis** [comment]
> I mean this file `.config/dfx/identity/${mainnet ? "default" : "default"}/identity.pem`, is it in your OS directory?

yup. I have it.

**Comment by b3hr4d** [comment]
Yes, I It should be. you can remove it for now, I'll fix it late.

**Comment by sadernalwis** [comment]
> > I mean this file `.config/dfx/identity/${mainnet ? "default" : "default"}/identity.pem`, is it in your OS directory?
> 
> yup. I have it.
> 
> btw, the exception is issued from ```ic-reactor```

I had the ```.pem``` file setup last time I had a successful build

**Comment by sadernalwis** [comment]
```bash
$ yarn deploy
yarn run v1.22.19
$ sh ./scripts/predeploy.sh
-e Building b3_system in ./backend/b3_system
    Finished release [optimized] target(s) in 0.02s
-e Size of b3_system.wasm: 3.8M
-e Generating Candid file for b3_system
-e Size of b3_system.did: 5.9K
-e Shrinking b3_system.wasm
-e Size of shrunk b3_system.wasm: 2.0M
Generating type declarations for canister b3_system:
  src/declarations/b3_system/b3_system.did.d.ts
  src/declarations/b3_system/b3_system.did.js
  src/declarations/b3_system/b3_system.did
-e Building b3_wallet in ./backend/b3_wallet
    Finished release [optimized] target(s) in 0.02s
-e Size of b3_wallet.wasm: 6.7M
-e Generating Candid file for b3_wallet
-e Size of b3_wallet.did: 14K
-e Shrinking b3_wallet.wasm
-e Size of shrunk b3_wallet.wasm: 4.7M
Generating type declarations for canister b3_wallet:
  src/declarations/b3_wallet/b3_wallet.did.d.ts
  src/declarations/b3_wallet/b3_wallet.did.js
  src/declarations/b3_wallet/b3_wallet.did
-e No Cargo.toml found in ./backend/lib. Skipping lib.
$ dfx deploy
Deploying all canisters.
Creating a wallet canister on the local network.
The wallet canister on the "local" network for user "default" is "bd3sg-teaaa-aaaaa-qaaba-cai"
Creating canisters...
Creating canister b3_system...
b3_system canister created with canister id: be2us-64aaa-aaaaa-qaabq-cai
Creating canister b3_wallet...
b3_wallet canister created with canister id: br5f7-7uaaa-aaaaa-qaaca-cai
Building canisters...
Executing 'true'
Executing 'true'
Executing 'true'
Executing 'true'
Executing 'scripts/predeploy.sh'
Building b3_system in ./backend/b3_system
    Finished release [optimized] target(s) in 0.02s
Size of b3_system.wasm: 3.8M
Generating Candid file for b3_system
Size of b3_system.did: 5.9K
Shrinking b3_system.wasm
Size of shrunk b3_system.wasm: 2.0M
Generating type declarations for canister b3_system:
  src/declarations/b3_system/b3_system.did.d.ts
  src/declarations/b3_system/b3_system.did.js
  src/declarations/b3_system/b3_system.did
Building b3_wallet in ./backend/b3_wallet
    Finished release [optimized] target(s) in 0.02s
Size of b3_wallet.wasm: 6.7M
Generating Candid file for b3_wallet
Size of b3_wallet.did: 14K
Shrinking b3_wallet.wasm
Size of shrunk b3_wallet.wasm: 4.7M
Generating type declarations for canister b3_wallet:
  src/declarations/b3_wallet/b3_wallet.did.d.ts
  src/declarations/b3_wallet/b3_wallet.did.js
  src/declarations/b3_wallet/b3_wallet.did
No Cargo.toml found in ./backend/lib. Skipping lib.
Executing 'scripts/predeploy.sh'
Building b3_system in ./backend/b3_system
    Finished release [optimized] target(s) in 0.02s
Size of b3_system.wasm: 3.8M
Generating Candid file for b3_system
Size of b3_system.did: 5.9K
Shrinking b3_system.wasm
Size of shrunk b3_system.wasm: 2.0M
Generating type declarations for canister b3_system:
  src/declarations/b3_system/b3_system.did.d.ts
  src/declarations/b3_system/b3_system.did.js
  src/declarations/b3_system/b3_system.did
Building b3_wallet in ./backend/b3_wallet
    Finished release [optimized] target(s) in 0.02s
Size of b3_wallet.wasm: 6.7M
Generating Candid file for b3_wallet
Size of b3_wallet.did: 14K
Shrinking b3_wallet.wasm
Size of shrunk b3_wallet.wasm: 4.7M
Generating type declarations for canister b3_wallet:
  src/declarations/b3_wallet/b3_wallet.did.d.ts
  src/declarations/b3_wallet/b3_wallet.did.js
  src/declarations/b3_wallet/b3_wallet.did
No Cargo.toml found in ./backend/lib. Skipping lib.
Installing canisters...
Installing code for canister b3_system, with canister ID be2us-64aaa-aaaaa-qaabq-cai
Installing code for canister b3_wallet, with canister ID br5f7-7uaaa-aaaaa-qaaca-cai
Error: Failed while trying to deploy canisters.
Caused by: Failed while trying to deploy canisters.
  Failed while trying to install all canisters.
    Failed to install wasm module to canister 'b3_wallet'.
      Failed during wasm installation call: The replica returned an HTTP Error: Http Error: status 413 Payload Too Large, content type "text/plain", content: Request 0x0291584fcf075faacb322fb86890dcfe7aa52a6de0ffde7512593a289ddd99e1 is too large. Message byte size 4927967 is larger than the max allowed 3670016.
error Command failed with exit code 255.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by b3hr4d** [comment]
I'll fix it now.

**Comment by b3hr4d** [comment]
BTW, the smaller is also large.
I should rewrite the B3Wallet soon.

**Comment by sadernalwis** [comment]
Thank you & really appreciate it.  🙏
Apologies for disturbing during holidays. 😊

**Comment by b3hr4d** [comment]
I have no holiday, I'm working all day, thanks for helping me :)

**Comment by sadernalwis** [comment]
> BTW, the smaller is also too large because of this `max allowed 3670016`. I should rewrite the B3Wallet soon.

but in the previous build like , with the old front end, I came across the same error [issue](https://github.com/B3Pay/b3-wallet/issues/27) I remember. but after fixing the other errors it disappeared.

What do you think the inconsistency is.?

**Comment by sadernalwis** [comment]
> I have no holiday, I'm working all day, thanks for helping me :)

Neither do I, don't really know what a holiday is anymore.. 😂

**Comment by b3hr4d** [comment]
> What do you think the inconsistency is.?

I have no Idea, also I got this error even after optimizing the wasm ` Message byte size 3977726 is larger than the max allowed 3670016.`. there should not be limit for local deployment!
I'll fix that tomorrow.

**Comment by sadernalwis** [comment]
> > What do you think the inconsistency is.?
> 
> I have no Idea, also I got this error even after optimizing the wasm ` Message byte size 3977726 is larger than the max allowed 3670016.`. there should not be limit for local deployment! I'll fix that tomorrow.

perfect!
we'll figure it out.

**Comment by sadernalwis** [comment]
Could it be [this](https://forum.dfinity.org/t/possible-bug-running-into-payload-too-large-error-even-though-my-canister-is-around-1-mb-in-size/19387/2?u=sadernalwis)

In that case we can write a ```deploy.sh``` to ```dfx canister install``` with ```.gz``` for each deployable canisters?

could it be a possible solution?

**Comment by sadernalwis** [comment]
```bash

    # Check if ic-wasm is installed before attempting to shrink the wasm file
    if command -v ic-wasm >/dev/null 2>&1; then
        # you can install ic-wasm via `cargo install ic-wasm` for smaller wasm files
        echo -e "${green}Shrinking $package.wasm${no_color}"
        mkdir -p "$optimised_target_dir"
        ic-wasm "$target_dir/$package.wasm" -o "$optimised_target_dir/$package.wasm" shrink
        # copy the candid file to the optimised target dir
        cp "$did_file" "$optimised_target_dir/$package.did"
        echo -e "Size of shrunk $package.wasm: $(ls -lh "$optimised_target_dir/$package.wasm" | awk '{print $5}')"
        echo -e "${green}Gunzipping $package.wasm${no_color}"
        gzip -c "$optimised_target_dir/$package.wasm" > "$optimised_target_dir/$package.wasm.gz"
        echo -e "Size of Gunzipped $package.wasm.gz: $(ls -lh "$optimised_target_dir/$package.wasm.gz" | awk '{print $5}')"
    else
        echo -e "${yellow}ic-wasm not found. Skipping shrinking $package.${no_color}"
    fi

```

**Comment by b3hr4d** [comment]
Nice, is gziped wasm run as canister with no issue?
I'll test it today.

**Comment by sadernalwis** [comment]
No. didn't test that yet.
have to double check scripts that should point to ```.gz```

**Comment by sadernalwis** [comment]
> Candid links doesn't work for both `b3_system` & `b3_wallet`

![image](https://github.com/B3Pay/b3-wallet/assets/44372160/4a5eee1d-8fac-4f00-a3b1-33f63e8f205c)

**Comment by sadernalwis** [comment]
Gunzipping optimized targets [PR](https://github.com/B3Pay/b3-wallet/pull/34)

**Comment by b3hr4d** [comment]
> Candid links doesn't work for both `b3_system` & `b3_wallet`

I guess this is not from gzip, It's because of ic-wasm. I had this issue before.

**Comment by b3hr4d** [comment]
Feel free to open new issue for each problem that you face, I'll fix it as soon as I have the time.

**Comment by sadernalwis** [comment]
A general ICP question. 3 actually. 😂
1. During local deployment, why is there a proxy hosting required. like `yarn dev`.
```bash
  /assets/icons/svg+xml/calendar.svg 1/1 (410 bytes) sha bbbcdcd7899b110d246b8d8473eaebe140e6f5cba266a36d936f5aba6a402bec 
Committing batch.
Committing batch with 500 operations.
Committing batch with 500 operations.
Committing batch with 190 operations.
Deployed canisters.
URLs:
  Frontend canister via browser
    frontend: http://127.0.0.1:4943/?canisterId=b77ix-eeaaa-aaaaa-qaada-cai
```

2. why isn't the `frontend` canister link not enough?
3. why doesn't the canister link load assets properly?

**Comment by b3hr4d** [comment]
> 1. During local deployment, why is there a proxy hosting required. like `yarn dev`.?

Actually do not need to deploy frontend at all, just deploy the `b3system` and you can run `yarn dev` and everything should work.

> 2\. why isn't the `frontend` canister link not enough?
Also this should work.
http://b77ix-eeaaa-aaaaa-qaada-cai.localhost:4943/

> 3\. why doesn't the canister link load assets properly?
Where did you face the issue, I guess we need to change to new proxy system for this, where we don't need proxy.

**Comment by sadernalwis** [comment]
For example, I'm trying to create a Ui patch to B3 starting with Vanilla (later Svelte, Vue etc).
[Build Automation Script](https://github.com/sadernalwis/Ui/blob/main/ICP/ICP.sh)
[Patch](https://github.com/sadernalwis/Ui/blob/main/Setup/Scripts/Python/DFX.py)

My problem is why when accessed Ui with;
```bash
URLs:
  Frontend canister via browser
    auth_client_demo_assets: http://127.0.0.1:4943/?canisterId=bw4dl-smaaa-aaaaa-qaacq-cai
  Backend canister via Candid interface:
    whoami: http://127.0.0.1:4943/?canisterId=by6od-j4aaa-aaaaa-qaadq-cai&id=b77ix-eeaaa-aaaaa-qaada-cai
```
1. ```http://localhost:4943/?canisterId=bw4dl-smaaa-aaaaa-qaacq-cai```   ➜  terrible load.
2. ```http://bw4dl-smaaa-aaaaa-qaacq-cai.localhost:4943```  ➜  less-terrible load.
3.  after ```yarn start``` using ```vite``` ```http://localhost:5173/```   ➜  good load!

My concern is regarding deploying on the main-net. is this going to be an issue?

**Comment by sadernalwis** [comment]
Ok. after further investigation looks like. loading is fine.
Some CSS issues, but fixable.
Btw, I can PR the [Build Automation Script](https://github.com/sadernalwis/Ui/blob/main/ICP/ICP.sh) if you like.

**Comment by b3hr4d** [comment]
> Build Automation Script

Thats interesting!
Did you check frontend build porcess on this repo? is this slow too?

> Btw, I can PR the [Build Automation Script](https://github.com/sadernalwis/Ui/blob/main/ICP/ICP.sh) if you like.

Can you please use `@ic-reactor/store` instead of using `agent` and `auth-client` directly? That would be helpful if you five me feedback on my new library.

**Comment by sadernalwis** [comment]
By 'terrible' loading, I wasn't very clear. I meant missing asset files specially CSS.
I didn't try the `frontend` build in like 4 or so days. it wasn't building then. Ill try again n get back.

The `ic-reactor` should be ok to work with other FE types right? (vanilla, svelte, vue et al)

by the new library did you mean the new `frontend`?

**Comment by b3hr4d** [comment]
> I didn't try the frontend build in like 4 or so days. it wasn't building then. Ill try again n get back.
I did today, its super fast :)

Yes `@ic-reactor/store` should work with everything.
Also, there is `@ic-reactor/core` that handle more use cases for vanilla and ....
Btw the docs are changed a lot, you can check the code if you are interested and make PR on that much appreciated.

> by the new library did you mean the new `frontend`?

The frontend is fine, ill working on that.
I want to use the `ic-reactor` library and give me feedback on that if you interested.

**Comment by sadernalwis** [comment]
Got it..
I have some ideas for the Documentation. 
Ill get to it.

**Comment by b3hr4d** [comment]
the example folder is updated, you can use it, and ask if you have any problem.
